packages/pyright-internal/src/common/configOptions.ts

```diff
@@ -2030,10 +2030,34 @@ export class ConfigOptions {
         configExtraPaths: Uri[]
     ): ExecutionEnvironment | undefined {
         try {
+            // If typeCheckingMode is specified for this execution environment,
+            // use it to generate the base diagnostic rule set. Otherwise, use
+            // the config-level diagnostic rule set.
+            let baseDiagnosticRuleSet = configDiagnosticRuleSet;
+            if (envObj.typeCheckingMode !== undefined) {
+                if (typeof envObj.typeCheckingMode === 'string') {
```

Comment:
pretty sure the `if (envObj.typeCheckingMode !== undefined)` statement is redundant